client/components/tinymce/plugins/after-the-deadline/plugin.js

```diff
@@ -0,0 +1,474 @@
+/*
+ * TinyMCE Writing Improvement Tool Plugin
+ * Author: Raphael Mudge (raffi@automattic.com), Andrew Duthie (andrew.duthie@automattic.com)
+ *
+ * http://www.afterthedeadline.com
+ *
+ * Distributed under the LGPL
+ *
+ * Derived from:
+ *	$Id: editor_plugin_src.js 425 2007-11-21 15:17:39Z spocke $
+ *
+ *	@author Moxiecode
+ *	@copyright Copyright (C) 2004-2008, Moxiecode Systems AB, All rights reserved.
+ *
+ *	Moxiecode Spell Checker plugin released under the LGPL with TinyMCE
+ */
+
+/**
+ * External dependencies
+ */
+import tinymce from 'tinymce/tinymce';
+import qs from 'querystring';
+import find from 'lodash/find';
+
+/**
+ * Internal dependencies
+ */
+import { getLocaleSlug, translate } from 'lib/mixins/i18n';
+import AtDCore from './core';
+import PreferencesActions from 'lib/preferences/actions';
+import PreferencesStore from 'lib/preferences/store';
+
+/**
+ * Module variables
+ */
+const SERVICE_HOSTNAME = 'service.afterthedeadline.com';
+const SERVICE_LOCALIZED_SUBDOMAINS = [ 'en', 'pt', 'de', 'es', 'fr' ];
```

Comment:
>Not knowing much about this service at all, should the button only be shown it the user's locale is available here?

Admittedly I'm following the precedent here, as [the Jetpack implementation](https://github.com/Automattic/jetpack/blob/618ea9652186ad975e63e96ec67962d9e918a6ee/modules/after-the-deadline/proxy.php#L83-L103) falls back to the non-localized subdomain when it cannot determine a supported locale.

There's also the question of whether we detect based on the user's locale, or the site's locale? I might imagine that we'd still want to show the option even if the user's specific locale is not supported, as they may be still be writing for an English site.